cpp/include/raft/distance/pairwise_distance_base.cuh

```diff
@@ -229,13 +286,36 @@ __global__ __launch_bounds__(
                                     EpilogueLambda epilog_op,
                                     FinalLambda fin_op) {
   extern __shared__ char smem[];
+  auto rowEpilog = [] __device__(IdxT starty) { return; };
 
   PairwiseDistances<useNorms, DataT, AccT, OutT, IdxT, Policy, CoreLambda,
-                    EpilogueLambda, FinalLambda, isRowMajor>
+                    EpilogueLambda, FinalLambda, decltype(rowEpilog),
+                    isRowMajor, writeOut>
     obj(x, y, m, n, k, lda, ldb, ldd, _xn, _yn, dOutput, smem, core_op,
-        epilog_op, fin_op);
+        epilog_op, fin_op, rowEpilog);
   obj.run();
 }
 
+template <typename P, typename IdxT>
+dim3 launchConfigGenerator(IdxT m, IdxT n) {
+  const auto numSMs = raft::getMultiProcessorCount();
+  // multiply by 2 as per launch bounds for pairwise dist kernels.
+  int minGridSize = numSMs * 2;
```

Comment:
instead of a constant factor of `2` how about computing this at runtime based on the occupancy of the kernel to be launched?

Reply:
done. making use of cudaOccupancyMaxActiveBlocksPerMultiprocessor for this purpose now.